addon/index.js

```diff
@@ -25,15 +30,19 @@ export default function(label, callback) {
     return callback;
   }
 
-  let lastReason;
+  let recentReasons = [];
   return function(reason) {
     // avoid reentrance and infinite async loops
-    if (reason === lastReason) {
-      lastReason = null;
+    const desc = reason.toString();
+    const pos = recentReasons.indexOf(desc);
+    if (pos !== -1) {
+      recentReasons = recentReasons
```

Comment:
This is basically making an LRU type of thing?

Reply:
Yeah, basically aiming to keep the same semantics we had before, but for multiple errors instead of just one. So wherever it appears in the list, remove it. Happy to change the data structure; this was the naïve/first-pass thing.